feat: allow to parse inline certs in the config file

ainghazal opened the pull request:
In order to support constructing a single config file it's convenient to be able to pass inline certs, as OpenVPN itself does.
These certs are identified by html-like tags: 

```
<ca>, <key>, <cert>
```

While working on this feature, I also:

- improved test coverage in unrelated parts
- made the ping stats also return more fields according to the latest spec.

bassosimone commented on extras/ping/ping.go:
```diff
@@ -248,7 +259,11 @@ func (p *Pinger) updateStatistics(pkt *Packet) {
 	p.PacketsRecv++
 	if p.RecordRtts {
-		p.rtts = append(p.rtts, pkt.Rtt)
+		p.rtts = append(p.rtts, pkt.Rtt.Seconds())
```
The `ping` tool uses milliseconds to represent RTTs, so I think we should do the same to reduce unneeded surprises into users of the tool. What do you think?

ainghazal replied:
it's true; I was already doing the conversion on the display. changed it to int64/ms.

ainghazal replied:
actually, I think I should keep a float64 with Microseconds / 1e3 to avoid truncating it, wdyt?

bassosimone commented on vpn/client.go:
```diff
@@ -83,18 +85,30 @@ func (c *Client) Start(ctx context.Context) error {
+		if err != nil {
+			return err
+		}
+		c.mux = mux
```
I am not super convinced by the fact that calling `Start` multiple times creates a new connection each time, however there is just a single `mux` factory that is created once. We should have a design discussion about this topic. My instinct is that either the whole of start is idempotent after the first call, or none of start is idempotent, but I am very surprised by a situation where just part of it are idempotent.

ainghazal replied:
incorporated these changes after our discussion today (I think I like #approach1 more, it makes more explicit that start is only supposed to be called once).

bassosimone commented on vpn/muxer_test.go:
```diff
@@ -50,21 +50,23 @@ func Test_newMuxerFromOptions(t *testing.T) {
+			 		tt.want.session.LocalSessionID[:],
+			 	)
+			 }
+			*/
```
I think you should delete commented out code or explain why we are going to need it later and what is the condition that will cause us to uncomment the code.

ainghazal replied:
there was a little TODO budired in there, but I agree - deleted the commented code for no, left only the TODO.

bassosimone commented on vpn/options.go:
```diff
@@ -390,14 +445,66 @@ func getOptionsFromLines(lines []string, dir string) (*Options, error) {
+	default:
+		return false
+	}
+}
```
Is the input file XML? If so, would it be possible to parse using `encoding/xml`?

ainghazal replied:
as discussed, it's not XML. the different blocks are identified by tags like:
```
<option>
---BEGIN BLOCK FOO---
...
---END BLOCK FOO
</option>
```
This is referred to as "inline file support" in openvpn's documentation. I will clarify the docs in minivpn about this.
I'm clarifying

bassosimone commented on vpn/options.go:
```diff
@@ -407,11 +514,14 @@ func hasElement(el string, arr []string) bool {
 
+// existsFile returns true if the file to which the path refers to exists.
 func existsFile(path string) bool {
 	_, err := os.Stat(path)
 	return !errors.Is(err, os.ErrNotExist)
```
```suggestion
	statbuf, err := os.Stat(path)
	return !errors.Is(err, os.ErrNotExist) && statbuf.Mode().IsRegular()
```

(You don't want this function to return true for an existing directory)

ainghazal replied:
fixed

bassosimone reviewed (commented):
We should have a chat about `Start`: I find the changes a bit confusing

ainghazal commented:
I have addressed the comments in the review and force-pushed to the branch for this MR.
Additionally, while writing documentation for the inline cert feature I realized that there was a potential LFI, that I've addressed at https://github.com/ooni/minivpn/pull/13/commits/1edac4e631698fb316f95af8ba87f482d40cd7b3

Do note that I've spend some time squashing and reordering different commits together: I think it does make sense to preserve the 5 distinct commits linearly in the git history in this case (instead of further squashing all this PR into a single commit, since it's touching quite different things altogether).

bassosimone commented on vpn/options.go:
```diff
@@ -1,17 +1,36 @@
 package vpn
 
+//
 // Parse VPN options.
```
```suggestion
// Parse VPN options.
//
```

ainghazal commented on README.md:
```diff
@@ -41,13 +41,41 @@ SPDX-License-Identifier: GPL-3.0-or-later
 file in `data/provider/config`. The [bootstrap script](https://github.com/ooni/minivpn/blob/main/scripts/bootstrap-provider)
 can be useful.
 
+
```
no good reason, fixed :)
